Hilla downgrades dependencies in a flow lit hybrid application when doing production build.

### Describe the bug

Hilla downgrades dependencies in a flow lit hybrid application when doing production build.
When running the application in dev mode the versions are kept as they are but when running a production build the dependencies are downgraded.

```
npm ERR! Invalid: lock file's @vaadin/a11y-base@24.3.2 does not satisfy @vaadin/a11y-base@24.3.0
npm ERR! Invalid: lock file's @vaadin/component-base@24.3.2 does not satisfy @vaadin/component-base@24.3.0
npm ERR! Invalid: lock file's @vaadin/field-base@24.3.2 does not satisfy @vaadin/field-base@24.3.0
npm ERR! Invalid: lock file's @vaadin/input-container@24.3.2 does not satisfy @vaadin/input-container@24.3.0
npm ERR! Invalid: lock file's @vaadin/lit-renderer@24.3.2 does not satisfy @vaadin/lit-renderer@24.3.0
npm ERR! Invalid: lock file's @vaadin/overlay@24.3.2 does not satisfy @vaadin/overlay@24.3.0
npm ERR! Invalid: lock file's @vaadin/vaadin-lumo-styles@24.3.2 does not satisfy @vaadin/vaadin-lumo-styles@24.3.0
npm ERR! Invalid: lock file's @vaadin/vaadin-material-styles@24.3.2 does not satisfy @vaadin/vaadin-material-styles@24.3.0
```

snippet from package.json after building for production.
```
      "@vaadin/a11y-base": "24.3.0",
      "@vaadin/accordion": "24.3.2",
```

### Expected-behavior

The versions should be the later versions.

### Reproduction

Run the hybrid app in dev mode and check that you get no version warnings.

Then build the project in production mode. The package.json will now contain downgraded versions of some libraries.

### System Info

Vaadin version 24.3.2 and hilla 2.5.5

--- a/scripts/prepare.sh
+++ b/scripts/prepare.sh
@@ -4,7 +4,7 @@
 # cannot use maven project.version here, as file generation happens before building maven
 version=`jq .version packages/ts/generator-typescript-core/package.json | cut -d '"' -f 2`
 # TODO: compute this number when we maintain multiple hilla branches
-branch=24.3.0
+branch=24.3
 
 # download needed files from vaadin/platform
 url=https://raw.githubusercontent.com/vaadin

